sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/ParDoTest.java

```diff
@@ -4085,7 +4085,8 @@ public void onTimer(OutputReceiver<Long> r) {
       ValidatesRunner.class,
       UsesStatefulParDo.class,
       UsesTimersInParDo.class,
-      UsesLoopingTimer.class
+      UsesLoopingTimer.class,
+      UsesStrictTimerOrdering.class
```

Comment:
Should `UsesLoopingTimer` imply `UsesStrictTimerOrdering` automatically?

Reply:
Well yes, but _any_ use of timers + state implies UseStrictTimerOrdering (since otherwise the GC timer could fire early and cleanup state). It just so happens that the particular bug I'm fixing mostly occurs during looping timers.

When it comes to test categories, I'm not sure how to make one imply the other. Ideally of course we should fix all runners, in which case we won't need any of these test categories.

Reply:
Agree.